views/pages/RegisterPage.jsx

```diff
@@ -26,47 +26,61 @@ const RegisterPage = ({ classes }) => {
   })
   const navigate = useNavigate()
   const handleFormSubmit = async (data) => {
-    switch (true) {
-      case data.password !== data.confirmPassword:
+    if (data.password !== data.confirmPassword) {
+      setErrors({
+        ...errors,
+        password: { error: true, message: 'passwords do not match' },
+      })
+
+      return
+    } else {
+      setErrors({
+        ...errors,
+        password: { error: false, message: '' },
+      })
+    }
```

Comment:
should we reset all errors in this case?

Reply:
In the case that there are still errors in the form and they clicked submit we should still display those

Reply:
but it looks like we're also going to call the API to sign up the user, which we probably don't want to do if there are form errors.

Reply:
[e2b0115](https://github.com/AMP-SCZ/dpdash/pull/416/commits/e2b011553eaee524d41911ef8a799d0b1fed909c)
I had to test it I thought the error state from the input would prevent it from submission. That is not the case.
Now we check that the form is clear of all errors before it can be submitted.